### DISCUSSION TITLE
RTC collaboration using CRDT and Yjs ?

### DISCUSSION THREAD

**Discussion by reyman** [outline/outline#2499] [Ideas]
Hi ! 
Thanks for all the work behind outline :)

When i see that you use prosemirror, i first think, "GREAT ! a wiki with full Real Time Collaboration possibility" !  
But after some test, i see this is not the case, when user1 edit a page, user2 on the same edit page need to reload...

There is a big cost to migrate prosemirror to use RTC full capacity, for example using CRDT and Yjs, like hackmd/googledoc/etc. permit : 
https://github.com/yjs/yjs and https://github.com/yjs/y-prosemirror
https://discuss.prosemirror.net/t/prosemirror-crdts/1190/8

There is also the possibility of using : https://www.hocuspocus.dev/ and really great tiptap prosemirror wrapper : https://www.tiptap.dev/  ?

I'm really interested to beta test and github sponsoring any new functionalities in this way.

Best Regards

**Comment by reyman** [comment]
Found my answer here : https://github.com/outline/outline/pull/1660
Great works ! I'm happy to see this is on the roadmap for a near future (+ other way to signing with OAuth2 :) ) 

Actually there is no open-source solution that give such RTC possibility + wiki possibilities , what you do with outline is game changing, well done !

**Reply by reyman** [reply]
One more challenge in RTC was to implement E2E encryption, like https://www.serenity.re/en/notes (front and back end are open source) and https://www.skiff.org/ (closed source)

**Reply by tommoor** [reply]
Thanks! I am actively contributing to Hocuspocus to add functionality needed to complete collab editing. RE: E2E encryption, it is not going to be possible with this project as fast server-side search is a priority.
